File: lab/src/main/java/com/lab/model/Vehicle.java
package com.lab.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.persistence.*;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String model;
    private String color;
    private double price;

    @ManyToOne
    @JoinColumn(name = "brand_id")
    private Brand brand;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private Owner owner;
}

File: lab-frontend/src/app/models/Vehicle.ts
import { Brand } from './Brand';
import { Owner } from './Owner';

export interface Vehicle {
  id: number;
  model: string;
  color: string;
  price: number;
  brand: Brand; // Ajuste para la relación con Brand
  owner: Owner; // Ajuste para la relación con Owner
}